### DISCUSSION TITLE
关于计算器的式子

### DISCUSSION THREAD

**Discussion by bsnmldb** [liusy58/NJU_AdvancedProgramming_SP22#18] [Q&A] [answered]
ppt上说”1×-1"这样的式子是不合法的，那么对于负号来说，一定要"1×(-1)"这样吗，那么"-1"就不合法，要"(-1)"吗这是关于负号的使用的合法性，应该要统一吧

**Comment by RicoloveFeng** [comment, marked as answer]
`1x-1`不合法是因为不能有两个连续的运算符，负号可以出现在表达式的开头或者左括号的后面，这里没有什么冲突吧

**Reply by bsnmldb** [reply]
主要是想把负号和它后面的数字看成一整块，一个单位

**Reply by RicoloveFeng** [reply]
总之规则就是上面说的那样，内部怎么处理由你来选择

**Reply by bsnmldb** [reply]
那“-1×1” "1+(-1×1)"应该合法吧

**Reply by RicoloveFeng** [reply]
是的
